pkg/apiserver/auditing/backend.go

```diff
@@ -0,0 +1,117 @@
+package auditing
+
+import (
+	"bytes"
+	"context"
+	"crypto/tls"
+	"encoding/json"
+	"fmt"
+	"k8s.io/client-go/listers/auditregistration/v1alpha1"
+	"k8s.io/klog"
+	"net/http"
+	"time"
+)
+
+const (
+	WaitTimeout = time.Second
+)
+
+type Backend struct {
+	lister          v1alpha1.AuditSinkLister
+	channelCapacity int
+	semCh           chan interface{}
+	cache           chan *EventList
+	client          http.Client
+	sendTimeout     time.Duration
+	waitTimeout     time.Duration
+}
+
+func NewBackend(channelCapacity int, cache chan *EventList, sendTimeout time.Duration, lister v1alpha1.AuditSinkLister) *Backend {
+
+	b := Backend{
+		semCh:           make(chan interface{}, channelCapacity),
+		channelCapacity: channelCapacity,
+		waitTimeout:     WaitTimeout,
+		cache:           cache,
+		sendTimeout:     sendTimeout,
+		lister:          lister,
+	}
+
+	b.client = http.Client{
+		Transport: &http.Transport{
+			TLSClientConfig: &tls.Config{
+				InsecureSkipVerify: true,
+			},
+		},
+		Timeout: b.sendTimeout,
+	}
+
+	go b.worker()
+
+	return &b
+}
+
+func (b *Backend) worker() {
+
+	for {
```

Comment:
how do we end this loop? There is only one condition to end this loop `event == nil`, but when program is gracefully terminated, how can we end this loop gracefully?